### PR TITLE
[KernelGen][Nvidia] Add hardshrink_backward operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `hardshrink_backward`. Computes the gradient of the Hard Shrink activation function with respect to its input
during backpropagation. The gradient is passed through where the input magnitude is
strictly greater than the threshold lambd, and is zero elsewhere.

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (15 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_hardshrink_backward.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.515 | 1.484 | 1.021 |
| 64, 64 | 0.006 | 0.006 | 1.034 |
| 4096, 4096 | 0.031 | 0.030 | 1.025 |
| 64, 512, 512 | 0.031 | 0.030 | 1.004 |
| 1024, 1024, 1024 | 1.889 | 1.876 | 1.007 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 3.369 | 3.519 | 0.958 |
| 64, 64 | 0.006 | 0.005 | 1.089 |
| 4096, 4096 | 0.053 | 0.053 | 1.005 |
| 64, 512, 512 | 0.054 | 0.054 | 1.002 |
| 1024, 1024, 1024 | 3.058 | 3.055 | 1.001 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 1073741824 | 1.522 | 1.465 | 1.039 |
| 64, 64 | 0.006 | 0.005 | 1.115 |
| 4096, 4096 | 0.031 | 0.030 | 1.022 |
| 64, 512, 512 | 0.031 | 0.030 | 1.018 |
| 1024, 1024, 1024 | 1.542 | 1.537 | 1.003 |

**Arithmetic Mean Speedup: 1.023x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.023 | Primary |
| Tianshu | — | — | — | — |
| Muxi | — | — | — | — |
| Ascend | — | — | — | — |
| Hygon | — | — | — | — |

## Files Changed
- `src/flag_gems/ops/hardshrink_backward.py`: Triton kernel implementation
- `tests/test_hardshrink_backward.py`: Accuracy test
- `benchmark/test_hardshrink_backward.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

**Review by bin913 — DISMISSED** [review]

File: src/flag_gems/ops/hardshrink_backward.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    self_c = self_t.contiguous()
+    out = torch.empty_like(grad_out_c)
+    _launch_hardshrink_backward_kernel(grad_out_c, self_c, out, lambd)
+    return out.reshape_as(grad_out)
+
+
+def hardshrink_backward_grad_input(
+    grad_out: torch.Tensor, self_t: torch.Tensor, lambd=0.5, *, grad_input=None
+):
+    logger.debug("GEMS HARD SHRINK BACKWARD.GRAD_INPUT")
```
Comment (bin913) [inline src/flag_gems/ops/hardshrink_backward.py]:
```suggestion
    logger.debug("GEMS HARDSHRINK_BACKWARD_GRAD_INPUT")
```

File: src/flag_gems/ops/hardshrink_backward.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            float(lambd),
+            # Simple 1D masked elementwise kernel; 1024 keeps blocks warp-aligned
+            # and saturates memory bandwidth without grid-size overhead.
+            BLOCK_SIZE=1024,
+            num_warps=4,
+        )
+
+
+def hardshrink_backward(grad_out: torch.Tensor, self_t: torch.Tensor, lambd=0.5):
+    logger.debug("GEMS HARD SHRINK BACKWARD")
```
Comment (bin913) [inline src/flag_gems/ops/hardshrink_backward.py]:
```suggestion
    logger.debug("GEMS HARDSHRINK_BACKWARD")
```

File: conf/operators.yaml [hunk 3]
```diff
@@ -6704,6 +6704,21 @@ ops:
...
+  for:
+  - hardshrink_backward
+  labels:
+  - aten
+  - pointwise
+  - KernelGen
+  kind:
+  - NeuralNetwork
+  stages:
+  - alpha: '5.4'
```
Comment (bin913) [inline conf/operators.yaml]:
```suggestion
  - alpha: '5.5.0'
```

**Review by bin913 — APPROVED** [review]
